File: core/src/main/java/pl/joboffer/job/security/service/RegisterService.java
package pl.joboffer.job.security.service;

import pl.joboffer.job.dto.company.CompanyRegisterDetails;
import pl.joboffer.job.dto.registerResponse.RegistrationResponse;
import pl.joboffer.job.dto.user.UserRegisterDetails;

public interface RegisterService {

  RegistrationResponse signup(UserRegisterDetails userRegisterDetails);

  RegistrationResponse companySignup(CompanyRegisterDetails companyRegisterDetails);
}

File: web/src/services/register/register.service.ts
import { Injectable } from '@angular/core';
import {RestService} from "../rest/rest.service";
import {catchError, Observable, throwError} from "rxjs";
import {UserRegisterDetails} from "../../app/dto/model/user/user-register-details";
interface RegistrationResponse {
  emailUsed: boolean;
}
@Injectable({
  providedIn: 'root'
})
export class RegisterService {
  private readonly registerUrl = 'auth/signup';
  constructor(private restService:RestService) { }

  register(userRegisterDetails: Omit<UserRegisterDetails, "repeatPassword">): Observable<RegistrationResponse> {
    return this.restService.post(this.registerUrl, userRegisterDetails).pipe(
      catchError(err => {
        console.error('Błąd HTTP:', err);
        return throwError(() => err);
      })
    );
  }
}
